process/transaction/transactionCostEstimator.go

```diff
@@ -46,10 +73,13 @@ func (tce *transactionCostEstimator) ComputeTransactionGasLimit(tx *transaction.
 	tx.GasPrice = 1
 
 	switch txType {
-	case process.MoveBalance, process.SCDeployment:
-		return tce.feeHandler.ComputeFee(tx).Uint64(), nil
+	case process.MoveBalance:
+		return tce.feeHandler.ComputeGasLimit(tx), nil
+	case process.SCDeployment:
+		gasLimit := uint64(len(tx.Data)) * (tce.storePerByteCost + tce.compilePerByteCost)
```

Comment:
for scDeployment and SCinvoking the base cost of the transaction has to be added. The base cost is normalTXCost + scExecution. So : sc.economicsFee.ComputeGasLimit(tx) + scdeployement cost or scinvoking cost

Reply:
done.,